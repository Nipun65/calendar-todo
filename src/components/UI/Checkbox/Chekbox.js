import styles from './Checkbox.module.css';

function Checkbox({ onClick, classes, checked }) {
  const handleClick = () => {
    onClick();
  };

  return (
    <input
      type="checkbox"
      checked={checked}
      onClick={handleClick}
      className={`${classes} ${styles.checkbox}`}
      readOnly
    />
  );
}
export default Checkbox;
