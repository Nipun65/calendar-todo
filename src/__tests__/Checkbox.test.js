import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Checkbox from '../components/UI/Checkbox/Chekbox';

afterEach(() => {
  cleanup();
});

it('should match the snapshot', () => {
  const tree = renderer.create(<Checkbox classes="is-inline" />).toJSON();
  expect(tree).toMatchSnapshot();
});
