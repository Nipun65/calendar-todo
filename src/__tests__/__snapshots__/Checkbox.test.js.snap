// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<input
  className="is-inline is-clickable"
  onClick={[Function]}
  readOnly={true}
  type="checkbox"
/>
`;
